### DISCUSSION TITLE
Security vulnerability by ghostscript

### DISCUSSION THREAD

**Discussion by chriskloe** [ImageMagick/ImageMagick#7411] [Help] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

Windows 10/11

### Description

Hi!

I am using parts of ImageMagick (e.g. compare) in an automated user-interface test of an application.
Two days ago a tool for detection of security vulnerabilities reported a finding in compare.exe (artifex_gpl_ghostscript_7.1.1.0).

Since that finding I updated my installation to ImageMagick 7.1.1-33 (before I had any 7.1.0).
Unfortunately the same vulnerability still has been reported. I tried to find the real ghostscript version in use but did not find any documentation.

Thus two questions to the commnity:
- which ghostscript version is actually used by ImageMagick (can't image it's a version from 2001)
- just in case: is there a way to update the dependency to the latest version, e.g. by replacing a dll?

Thanks for any kind of support!

BR,
Christoph

**Comment by dlemstra** [comment, marked as answer]
We don't bundle the Ghostscript library with our project. That  library requires a paid license for commercial use so we cannot bundle it. We only search the registry to find the location where it is installed. This look like a bug / false positive in the tool that you are using.

**Comment by guldil** [comment]
Same here with Microsoft Defender for Business. "compare.exe" is the tool detected with artife gpl_ghostscript (Critical).

I made a report.

**Comment by mallen1972** [comment]
Hi, we are seeing the same vulnerability flagged by Microsoft Defender for an .exe file called convert.exe which is part of the Adobe Photoshop suite. I believe this is one of your .exe's. Are you able to confirm if that it is and if so if it is likely to contain the Ghostscript library. Adobe have been less than helpful with clarification.  We are seeing ImageMagick version 6.9.9.0.
Many thanks
Mark

**Reply by dlemstra** [reply]
>We don't bundle the Ghostscript library with our project. That library requires a paid license for commercial use so we cannot bundle it. We only search the registry to find the location where it is installed. This look like a bug / false positive in the tool that you are using.

**Reply by pstakuu** [reply]
Did ImageMagick ever use Ghostscript? As in, in version 6.9.9?
![image](https://github.com/user-attachments/assets/76be414c-dc40-4926-9e55-41331d79ebf5)

**Comment by fmw42** [comment]
Photoshop does not use Imagemagick convert.exe.  Might it be the windows convert.exe?

**Reply by dlemstra** [reply]
They are using ImageMagick. I am still waiting for them to send us free licenses for Photoshop.....

**Reply by fmw42** [reply]
Dirk, what are they using Imagemagick to do or in what software?

**Reply by dlemstra** [reply]
I have no clue what they are using it for but it have seen it installed on Winows with Photoshop.

**Reply by pstakuu** [reply]
![image](https://github.com/user-attachments/assets/85befe93-7c4c-477b-981c-0cf6bfc51371)
Adobe definitely does.  This is in the Program Files for Adobe

**Comment by snibgo** [comment]
Fairly obviously, try executing the program, eg:
```
convert /?
convert -version
```

**Reply by pstakuu** [reply]
![image](https://github.com/user-attachments/assets/bfc93f4a-d6e7-443c-bd45-1d0124e2d6e1)
Any other ideas? 
To me this has gotta be a combo of things: 
Back in time when ImageMagick was on version 6.9.9.0, they used the GhostScript library.  Adobe packages that version with their Photoshop product.  ImageMagick has since stopped using the library.  So Adobe needs to use an updated version of ImageMagick?

ImageMagick - check out what Adobe puts on our computers:
![image](https://github.com/user-attachments/assets/69cff1d4-dedf-4d49-8969-dd2f40a647e3)

**Reply by dlemstra** [reply]
We are not Adobe so you are asking the question at the wrong place.
